Testing/C2DTests.cs

```diff
@@ -41,12 +44,34 @@ public async Task SendC2DMessages()
             // Act
             consumer.ConnectConsumer();
             await consumer.SendCloudToDeviceMessageAsync("test");
-            while (!flag)
-            {
+            while (!flag) { }
 
-            }
+            // Assert
             Assert.IsTrue(flag);
-            //Assert.Pass();
+        }
+
+        [Test]
+        public async Task SendD2CWithRetainFlagTrue()
+        {
+            // Arrange
+            await device.ConnectDevice();
+
+            var retainFlag = true;
+            var payload = Guid.NewGuid().ToString();
+
+            // Act
+            // send D2C with retain flag set to true
+            await device.SendDeviceToCloudMessageAsync(payload, retainFlag);
+
+            using var cancellationSource = new CancellationTokenSource();
+
+            // Assert - verify received + mqtt-retain set to true
+            cancellationSource.CancelAfter(3000);
+            var recConsumer = new ReceiverConsumer(configuration);
```

Comment:
I'd move this declaration out to the constructor with the others

Reply:
@EliiseS Did you mean "out to the Setup() method with the others"?

Reply:
Yes, exactly 😄 Setup, constructor, pish, posh